mlir/lib/Dialect/Vector/Transforms/VectorTransforms.cpp

```diff
@@ -1152,8 +1152,71 @@ struct FoldI1Select : public OpRewritePattern<arith::SelectOp> {
   }
 };
 
-// Drop inner most contiguous unit dimensions from transfer_read operand.
-class DropInnerMostUnitDims : public OpRewritePattern<vector::TransferReadOp> {
+/// Returns the number of dims can be folded away from transfer ops. It returns
+/// a failure if strides and offsets can not be resolved.
```

Comment:
1. How are such "dims" defined? Perhaps add examples?
2. "It returns a failure if strides and offsets can not be resolved." - IMHO that's an implementation detail that can be skipped from this higher level description.

Btw - in what cases are strides not resolved? Is it something that we could test?

Reply:
I think we are not able to test that, it is mostly added for safety (IMHO). The logic was copied from the previous implementation, and I don't want to update other conditions in a PR.